src/gluonts/model/deepvar/_estimator.py

```diff
@@ -0,0 +1,338 @@
+# Standard library imports
+from typing import List, Optional, Callable, Tuple
+
+# Third-party imports
+import re
+import pandas as pd
+import numpy as np
+from mxnet.gluon import HybridBlock
+
+# First-party imports
+from gluonts.core.component import validated
+from gluonts.distribution import DistributionOutput, StudentTOutput
+from gluonts.model.estimator import GluonEstimator
+from gluonts.model.predictor import Predictor, RepresentableBlockPredictor
+from gluonts.support.util import copy_parameters
+
+from gluonts.time_feature import TimeFeature
+from gluonts.dataset.field_names import FieldName
+
+from gluonts.trainer import Trainer
+from gluonts.transform import (
+    AddObservedValuesIndicator,
+    AddTimeFeatures,
+    AsNumpyArray,
+    Chain,
+    ExpectedNumInstanceSampler,
+    InstanceSplitter,
+    SetFieldIfNotPresent,
+    Transformation,
+    VstackFeatures,
+    ExpandDimArray,
+    TargetDimIndicator,
+    CDFtoGaussianTransform,
+    cdf_to_gaussian_forward_transform,
+    RenameFields,
+)
+
+# Relative imports
+from ._network import DeepVARPredictionNetwork, DeepVARTrainingNetwork
+
+
+class FourierDateFeatures(TimeFeature):
+    @validated()
+    def __init__(self, freq: str) -> None:
+        super().__init__()
+        # reocurring freq
+        freqs = [
+            "month",
+            "day",
+            "hour",
+            "minute",
+            "weekofyear",
+            "weekday",
+            "dayofweek",
+            "dayofyear",
+            "daysinmonth",
+        ]
+
+        assert freq in freqs
+        self.freq = freq
+
+    def __call__(self, index: pd.DatetimeIndex) -> np.ndarray:
+        values = getattr(index, self.freq)
+        num_values = max(values) + 1
+        steps = [x * 2.0 * np.pi / num_values for x in values]
+        return np.vstack([np.cos(steps), np.sin(steps)])
+
+
+def get_granularity(freq_str: str) -> Tuple[int, str]:
```

Comment:
Don't we have this code elsewhere as well?

Reply:
Not exactly, AFAIK. 

One can use the `to_offset` function from pandas, though, to produce the same thing.

Reply:
Not exactly. One can use the `to_offset` function from pandas to do the same thing, though.